executor/index_lookup_merge_join.go

```diff
@@ -0,0 +1,735 @@
+	defer func() {
+		if r := recover(); r != nil {
+			err := errors.Errorf("%v", r)
+			logutil.Logger(context.Background()).Error("outerMergeWorker panic in the recoverable goroutine", zap.Error(err))
+			task := &lookUpMergeJoinTask{
+				results: make(chan *chunk.Chunk, 1),
+				doneErr: make(chan error, 1),
+				done:    true,
+			}
+			task.doneErr <- err
+			omw.pushToChan(ctx, task, omw.resultCh)
+		}
+		close(omw.resultCh)
+		close(omw.innerCh)
+		wg.Done()
+	}()
+	for {
+		task, err := omw.buildTask(ctx)
+		if err != nil {
+			task.doneErr <- err
+			omw.pushToChan(ctx, task, omw.resultCh)
+			return
+		}
+		if task == nil {
+			return
+		}
+
+		if finished := omw.pushToChan(ctx, task, omw.innerCh); finished {
+			return
+		}
+
+		if finished := omw.pushToChan(ctx, task, omw.resultCh); finished {
+			return
+		}
+	}
+}
+
+func (omw *outerMergeWorker) pushToChan(ctx context.Context, task *lookUpMergeJoinTask, dst chan<- *lookUpMergeJoinTask) bool {
+	select {
+	case <-ctx.Done():
+		return true
+	case dst <- task:
+	}
+	return false
+}
+
+// buildTask builds a lookUpMergeJoinTask and read outer rows.
+// When err is not nil, task must not be nil to send the error to the main thread via task
+func (omw *outerMergeWorker) buildTask(ctx context.Context) (*lookUpMergeJoinTask, error) {
+	newFirstChunk(omw.executor)
+	task := &lookUpMergeJoinTask{
+		results:     make(chan *chunk.Chunk, 1),
+		doneErr:     make(chan error, 1),
+		outerResult: newFirstChunk(omw.executor),
+	}
+	task.memTracker = memory.NewTracker(stringutil.MemoizeStr(func() string { return fmt.Sprintf("lookup join task %p", task) }), -1)
+	task.memTracker.AttachTo(omw.parentMemTracker)
+
+	omw.increaseBatchSize()
+	if omw.lookup.isOuterJoin { // if is outerJoin, push the requiredRows down
+		requiredRows := int(atomic.LoadInt64(&omw.lookup.requiredRows))
+		task.outerResult.SetRequiredRows(requiredRows, omw.maxBatchSize)
+	} else {
+		task.outerResult.SetRequiredRows(omw.batchSize, omw.maxBatchSize)
+	}
+
+	task.memTracker.Consume(task.outerResult.MemoryUsage())
+	for !task.outerResult.IsFull() {
+		err := Next(ctx, omw.executor, omw.executorChk)
+		if err != nil {
+			return task, err
+		}
+		if omw.executorChk.NumRows() == 0 {
+			break
+		}
+
+		oldMemUsage := task.outerResult.MemoryUsage()
+		task.outerResult.Append(omw.executorChk, 0, omw.executorChk.NumRows())
+		newMemUsage := task.outerResult.MemoryUsage()
+		task.memTracker.Consume(newMemUsage - oldMemUsage)
+	}
+	if task.outerResult.NumRows() == 0 {
+		return nil, nil
+	}
+	if omw.outerMergeCtx.needOuterSort == false {
+		c := newFirstChunk(omw.executor)
+		var rows []chunk.Row
+		for i := 0; i < task.outerResult.NumRows(); i++ {
+			rows = append(rows, task.outerResult.GetRow(i))
+		}
+		sort.Slice(rows, func(i, j int) bool {
+			for id, joinKey := range omw.joinKeys {
+				cmp, _, err := omw.compareFuncs[id](omw.ctx, joinKey, joinKey, rows[i], rows[j])
+				terror.Log(err)
```

Comment:
Will ignoring this error make the result wrong?

Reply:
It's as same as `index_lookup_join`. Look at https://github.com/pingcap/tidb/blob/2ea24e02015c92cbcc0595df5554167ae4494339/executor/index_lookup_join.go#L563